pkg/client/run.go

```diff
@@ -84,6 +85,14 @@ func (c *SonobuoyClient) RunManifest(cfg *RunConfig, manifest []byte) error {
 		if err != nil {
 			return errors.Wrap(err, "could not get object namespace")
 		}
+
+		// The namespace value is required if Wait is enabled in order to check the status.
+		// It may not be available within the RunConfig but we can retrieve it from the
+		// namespace of objects within the manifest.
+		if createdNamespace == "" && namespace != "" {
+			createdNamespace = namespace
```

Comment:
Why not just use `namespace`? I'm not clear on what `createdNamespace` adds since by default it is the empty string, and it is set to `namespace` if it isn't the empty string. So won't it just always match `namespace`?

Reply:
This loop is iterating over all the resources in the yaml output. Some of those are cluster-scoped (e.g. [ClusterRole](https://github.com/vmware-tanzu/sonobuoy/blob/master/pkg/templates/manifest.go#L51-#L71) which means that they do not have an associated namespace. If the yaml was edited to have these resources be the last ones to be iterated over, namespace would be empty at the end of the loop. You can see this in the logs from Sonobuoy run, some of the resources it logs don't have the namespace field set in the log. Storing the first non empty namespace was an attempt to combat that. This approach isn't fool proof, but seemed good enough for now as it avoids the need to fully parse the yaml and find the Namespace resource, or the contents of the config map (we do this in gen_test).